File: apps/ios-native/MemoryWiki/Views/BrandLoader.swift
// BrandLoader — the canonical loading affordance. Replaces
// ProgressView / shimmer skeletons across the app so every wait
// state reads as the brand instead of UIKit chrome.
//
// Animated morph blob (the same artwork the auth hero uses) over
// a faint mono "LOADING" caption. Compact variant for inline use
// (timeline empty state, doc body in flight); full-screen variant
// for boot.

import SwiftUI

struct BrandLoader: View {
    enum Variant {
        case inline       // bigger blob + caption — empty list / body in flight
        case full         // largest blob + caption — app boot
    }
    var variant: Variant = .inline
    var caption: String = "LOADING"

    // Larger sizes per the brand-loader feedback — the previous
    // 56/96pt blobs felt timid; bump for presence.
    private var size: CGFloat { variant == .full ? 140 : 96 }

    var body: some View {
        // Centre vertically + horizontally INSIDE whatever space
        // the caller hands us. Spacers + maxHeight: .infinity
        // means the blob+caption land in the middle of the
        // available content area — header / filter strip stays
        // put at the top, only the loader moves to centre.
        VStack(spacing: 18) {
            Spacer(minLength: 0)
            AnimatedBlob(size: size, theme: .dark)
            Text(caption)
                .font(Brand.mono(size: 10, weight: .medium))
                .tracking(1.4)
                .foregroundStyle(Brand.textFaint)
            Spacer(minLength: 0)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .accessibilityElement(children: .ignore)
        .accessibilityLabel("Loading")
    }
}

#Preview {
    ZStack {
        Brand.background.ignoresSafeArea()
        BrandLoader()
    }
}

File: apps/android-native/app/src/main/kotlin/wiki/memory/memorywiki/ui/components/BrandLoader.kt
/*
 * BrandLoader — canonical loading affordance. Port of iOS
 * MemoryWiki/Views/BrandLoader.swift.
 *
 * The brand blob (same SMIL-animated SVG the splash + auth hero
 * use) over a tracked mono "LOADING" caption in JetBrains Mono.
 * Replaces every full-screen CircularProgressIndicator so wait
 * states read as the brand instead of Material chrome.
 *
 * Variants:
 *   Inline — 96dp blob, list / body in flight
 *   Full   — 140dp blob, app boot
 *
 * Sits inside a fill-the-available-space Box so the caller can
 * give it any frame (e.g. .fillMaxSize() or a fixed minHeight).
 */

package wiki.memory.memorywiki.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import wiki.memory.memorywiki.ui.theme.Brand
import wiki.memory.memorywiki.ui.theme.BrandType

enum class BrandLoaderVariant { Inline, Full }

@Composable
fun BrandLoader(
    variant: BrandLoaderVariant = BrandLoaderVariant.Inline,
    caption: String = "LOADING",
    modifier: Modifier = Modifier,
) {
    val sizeDp = if (variant == BrandLoaderVariant.Full) 140 else 96
    Box(
        modifier = modifier.fillMaxSize(),
        contentAlignment = Alignment.Center,
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(18.dp),
        ) {
            BrandBlob(sizeDp = sizeDp)
            Text(
                caption,
                style = BrandType.mono(10, FontWeight.Medium).copy(letterSpacing = 1.4.sp),
                color = Brand.TextFaint,
            )
        }
    }
}
